packages/sdk/client-protocol/src/service-definitions.ts

```diff
@@ -50,6 +51,9 @@ export type ClientServices = {
 export interface ClientServicesProvider {
   descriptors: ServiceBundle<ClientServices>;
   services: Partial<ClientServices>;
+  // TODO(wittjosiah): Should this be required?
+  //  If reset happens in the shell this is required to communicate the reset client that the reset happen and restart.
+  terminated?: Event<void>;
```

Comment:
I think we can make this event required for all providers.
```suggestion
  /**
   * The connection to the services provider was termintaed.
   */
  closed: Event<Error | undefined>;
```